test/hybrid.jl

```diff
@@ -107,3 +112,60 @@ end
 
     @test norm(U .- sol.u[end]) ≤ 5e-5
 end
+
+@testset "1D SE, 1D FV Extruded Domain ∇ ODE Solve diagonally" begin
+
+    # Advection equation in Cartesian domain with
+    # uₕ = (cₕ, 0), uᵥ = (0, cᵥ)
+    # ∂ₜf + ∇ₕ⋅(uₕ * f) + ∇ᵥ⋅(uᵥ * f)  = 0
+    # the solution translates diagonally to the top right and
+    # at time t, the solution is f(x - cₕ * t, y - cᵥ * t)
+    # here cₕ == cᵥ == 1, integrate t == 2π or one full period.
+    # This is only correct if the solution is localized in the vertical
+    # as we don't use periodic boundary conditions in the vertical.
+    #
+    # NOTE: the equation setup is only correct for Cartesian domains!
+
+    hv_center_space, hv_face_space = hvspace_2D((-1, 1), (-1, 1))
+
+    function rhs!(dudt, u, _, t)
+        h = u.h
+        dh = dudt.h
+
+        # vertical advection no inflow at bottom 
+        # and outflow at top
+        Ic2f = Operators.InterpolateC2F(top = Operators.Extrapolate())
+        divf2c = Operators.DivergenceF2C(
+            bottom = Operators.SetValue(Geometry.Cartesian13Vector(0.0, 0.0)),
+        )
+        # only upward advection
+        @. dh = -divf2c(Ic2f(h) * Geometry.Cartesian13Vector(0.0, 1.0))
+
+        # only horizontal advection
+        hdiv = Operators.Divergence()
+        @. dh += -hdiv(h * Geometry.Cartesian1Vector(1.0))
+        Spaces.weighted_dss!(dh)
+
+        return dudt
+    end
+
+    # initial conditions
+    h_init(x_init, z_init) = begin
+        coords = Fields.coordinate_field(hv_center_space)
+        h = map(coords) do coord
+            exp(-((coord.x + x_init)^2 + (coord.z + z_init)^2) / (2 * 0.2^2))
+        end
+
+        return h
+    end
+
+    using OrdinaryDiffEq
+    U = Fields.FieldVector(h = h_init(0.5, 0.5))
+    Δt = 0.01
+    t_end = 1.0
+    prob = ODEProblem(rhs!, U, (0.0, t_end))
+    sol = solve(prob, SSPRK33(), dt = Δt)
+
+    h_end = h_init(0.5 - t_end, 0.5 - t_end)
+    @test norm(h_end .- sol.u[end].h) ≤ 5e-2
```

Comment:
should we check the error of the vertical / horizontal only components individually as well as the overall 2D solution?